### ISSUE TITLE
ValueError: deflate payloads are not supported

### ISSUE THREAD

**Issue by AmitGamingKing** [tgalal/yowsup#2671]
I'm using https://github.com/danielcardeenas/whatsapp-framework and after just starting script I'm facing following issue
```
[Whatsapp] Mac started

W 2019-04-24 00:11:54,612 yowsup.layers.protocol_ib.layer - ignoring edge_routing ib node for now
Traceback (most recent call last):
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/network/dispatcher/dispatcher_socket.py", line 42, in connectAndLoop
    self.connectionCallbacks.onRecvData(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/network/layer.py", line 82, in onRecvData
    self.receive(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/network/layer.py", line 85, in receive
    self.toUpper(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/noise/layer_noise_segments.py", line 39, in receive
    self.toUpper(bytes(data))
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/noise/layer.py", line 149, in receive
    self._flush_incoming_buffer()
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/noise/layer.py", line 137, in _flush_incoming_buffer
    self.toUpper(self._wa_noiseprotocol.receive())
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/__init__.py", line 79, in toUpper
    self.__upper.receive(data)
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/coder/layer.py", line 19, in receive
    node = self.reader.getProtocolTreeNode(bytearray(data))
  File "/home/amitgamingking/.local/lib/python3.6/site-packages/yowsup/layers/coder/decoder.py", line 11, in getProtocolTreeNode
    raise ValueError("deflate payloads are not supported")
ValueError: deflate payloads are not supported
E 2019-04-24 00:11:55,288 yowsup.layers.network.dispatcher.dispatcher_socket - deflate payloads are not supported
```
Running on latest Yowsup V3, all requirements are installed...

However, it is running fine on Terminal Login I can able to send messages but I can't send media files on terminal it is showing me error ```yowsup.demos.cli.layer - Request upload for file /path/to/pic.png for 91XXXXXXXXXX@s.whatsapp.net failed``` if I do ``/image send <number> <location>``

Kindly help me

**Comment by AmitGamingKing** [comment]
@tgalal please help me how to solve this thing? At least tell me why this is raising Exception?

**Comment by tgalal** [comment]
@AmitGamingKing you are receiving a message encoded in a format yowusp does not support yet. Do you have information about what you are receiving? Like what kind of message (notification/text/ audio/video ..etc), or from which kind of device as it sent. This information would be helpful.
What also would be helpful is a dump of the received bytes right before they are decoded. An alternative would be that you send me your phone number and static keypair in a private message/ email and I'll try to get this data myself.

**Comment by AmitGamingKing** [comment]
I'm not receiving any kind of message I'm getting this error before receiving any message means after just starting the script! I'll send you credentials shortly...

**Comment by AmitGamingKing** [comment]
@tgalal please check your mail I sent it to you :)

**Comment by tgalal** [comment]
fixed in [v3.1.0](https://github.com/tgalal/yowsup/releases/tag/v3.1.0)
